src/octoprint/server/api/files.py

```diff
@@ -192,6 +192,16 @@ def sanitize(storage, path, filename):
 
         exists = fileManager.file_exists(storage, sanitized)
         if exists:
+            exists_file = [
+                gcode_file
+                for gcode_file in _getFileList(storage)
+                if gcode_file["name"] == filename
+            ]
+            if len(exists_file) > 0:
+                exists_size = exists_file[0]["size"]
+                exists_date = exists_file[0]["date"]
+            else:
+                abort(400, description="Not exists file found to retrieve information")
```

Comment:
Wording is a bit strange on this description. Maybe something more like `File not found to retrieve additional information`. It would probably be a rare race condition of the file being deleted and getting into this else block, but just in case.